### DISCUSSION TITLE
Why deploy Argo components (Workflows, ArgoCD, Events, Rollouts) in different namespaces?

### DISCUSSION THREAD

**Discussion by jakepage91** [argoproj/argo-cd#19111] [Q&A] [answered]
I'm looking to stand up the full Argo ecosystem in a test cluster to try out the full Argo flavoured CICD system. In the Argo docs each component is to be deployed in it's own namespace:

- ArgoCD -> `argocd` namespace
- Argo Workflows -> `argo` namespaces
- Argo Events -> `argo-events` namespace
- Argo Rollouts -> `argo-rollouts` namespace

Why is this?

Wouldn't it make more sense to deploy them all in an `argo-system` ns for example?

If anybody has any experience deploying the different components to a single common namespace I would love to hear about your experience.

**Comment by ChristianCiach** [comment, marked as answer]
There is nothing stopping you from deploying all these products into the same namespace, but why should you? It's generally better to separate applications into their own namespace, for easier removal, simplier network-policies, RBAC...

All these products have nothing in common except for their development under the Argo umbrella. Surely you wouldn't deploy all the products made by Google into a "google-system" namespace?

**Reply by jakepage91** [reply]
Got it. Appreciate your thoughts. 

Initially I was under the impression that the tools were more tighyly coupled (I thought Argo Events depended always on Argo Workflows), I got this idea from this segment of the `Getting started` section: 

>We are going to set up a sensor and event-source for webhook. The goal is to trigger an Argo workflow upon an HTTP Post request.

> Note: You will need to have [Argo Workflows](https://argoproj.github.io/argo-workflows/) installed to make this work. The Argo Workflow controller will need to be configured to listen for Workflow objects created in argo-events namespace. (See [this](https://github.com/argoproj/argo-workflows/blob/master/docs/managed-namespace.md) link.)

I was under the impression that the only possible Argo event target was Argo Workflows but upon further reading I see that's not the case and that each tool can really be considered a standalone components and used independently from the others. 

That being the case it doesn't make much sense to group them together in the same namespace. Thanks @ChristianCiach

**Reply by ChristianCiach** [reply]
I see where you are coming from. At first I also thought that Argo Events and Workflows were tightly coupled, because Argo Workflows comes with a UI that visualizes Argo Events objects (EventSources, Sensors and TriggerTemplates). But I guess instead of "tightly coupled" we should more think of it as "plays nicely together".

**Reply by jakepage91** [reply]
Exactly!
